### DISCUSSION TITLE
How to pass-through ENV variables with Webpack?

### DISCUSSION THREAD

**Discussion by cjayyy** [vercel/turborepo#7588] [Help]
### Summary

Hi 👋 

I am trying to figure out how to pass environment variables through Webpack build into runtime but don't(?) cache it? Honestly I don't really know what exactly I need regarding caching/not-caching... 

My situation: 
I am building a Webpack app via Turborepo. This app needs `VERCEL_ENV` variable which is being "routed" via `webpack.DefinePlugin`. Since the app needs `VERCEL_ENV` only during runtime (condition is evaluated at the runtime), it is not necessary to invalidate cache based on this variable, but since `webpack.DefinePlugin` runs in build time it cannot be loaded from cache. 

I guess I am missing some fundamental concept here, but I am not able to find it. What could be some of the possible solutions? 

Thank you 

### Additional information

```js
// webpack.config.js (build)
...

plugins: [
      new webpack.DefinePlugin({
        'process.env': {
          VERCEL_ENV: JSON.stringify(process.env.VERCEL_ENV)
        }
      })
    ]

...
```

```js
// app (runtime)
...

const runtimeListener = () => {
  if (process.env.VERCEL_ENV == 'production') {
    return;
  }
...
}

...
```

**Comment by anthonyshew** [comment]
Can you provide a full reproduction?
